### ISSUE TITLE
Time for a Person Record Index Regen?

### ISSUE THREAD

**Issue by PatientRowan** [historyforge/historyforge#178]
Eve and I have been noticing small lapses in the person record index.  A name we know we have created a person record for does not show up in the index, or shows up in a mysterious order.  I have found myself refreshing the search, which does not help.  I should have screen-shot these occurrences, but like Eve, I have instead been pressing to get my task done, and snuck up on the missing record in other ways.
As you know, I don’t know what I’m talking about, but in previous systems I’ve worked in, when complaints of glitches like these got to a certain pitch, the systems office’s solution was to run an index regeneration.

**Comment by dfurber** [comment]
I rebuilt the index. See if it helped?

**Comment by dfurber** [comment]
And a button is on the way on Admin menu that will let you do it yourself.

**Comment by PatientRowan** [comment]
I clicked on Rebuild Index to see what the dialog box might look like, and instead, it just did it!  In no time flat, I expect because you’d just done it last night?  It seems a little too easy, like it’s a placebo button, or something.
Thank you very much.  All seems well.

**Comment by dfurber** [comment]
The reason it is quick is that there really aren't that many records, and all that needs to be indexed lies within the record. If we had millions of census records and a complicated system of tables and joins, it would take longer.

**Comment by PatientRowan** [comment]
On Sunday morning, after you did your rebuild and maybe or maybe not after I did my accidental rebuild, I checked on something I do periodically: I got All People up and used the sorting arrow to put them in birth year order.  IIRC, there were two records that impossibly early dates, like in the 1400s.  I fixed them, as I always do, and I didn’t screenshot them.
Here is a screenshot I have been sitting on for a while that shows that an automated update had messed with the birth year:
![DAD207A5-6E6E-4B2B-947D-552DA9103AA6](https://user-images.githubusercontent.com/114665933/226347260-846624b2-470f-4374-8f5f-2635eb79e06d.jpeg)

I’m totally fine with occasionally fixing a handful of records.  Two questions, though:
(1) My impression is maybe these impossible dates happen as a result of rebuilding the index?
(2) If we UNCHECK the “Estimated?” box on the person record (citing our source for the birth year in the Description field), does that protect the date from getting messed with?
